### PR TITLE
Add PHP 7.2.0-alpha.1 core formula (php72)

### PR DESCRIPTION
**Pull request by romainneutron**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-php/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)? 
 -> No it says `* Stable version URLs should not contain alpha` and it is normal

-----

### CODE REVIEW

**Comment by romainneutron** [comment]
I'm currently checking all extensions embedded in this PR. Please give me some time to remove unsupported extension or update the one that need update. I'll provide feedback here

**Comment by romainneutron** [comment]
Okay, everything in this PR has been installed and tested using `brew audit --strict --online <formula>`. It's alright except for the PHP download URL that contains `alpha` (but it seems normal)

**Comment by romainneutron** [comment]
Pull request is ready !

**Comment by romainneutron** [comment]
And here's the final complete list of added formulae:

`php72`, `php72-amqp`, `php72-apcu`, `php72-apcu-bc`, `php72-ast`, `php72-blitz`, `php72-couchbase`, `php72-ev`, `php72-event`, `php72-gearman`, `php72-geoip`, `php72-geos`, `php72-gmagick`, `php72-gmp`, `php72-grpc`, `php72-hprose`, `php72-igbinary`, `php72-imagick`, `php72-intl`, `php72-libsodium`, `php72-lua`, `php72-lz4`, `php72-lzf`, `php72-mailparse`, `php72-maxminddb`, `php72-mecab`, `php72-memcached`, `php72-mongodb`, `php72-msgpack`, `php72-mustache`, `php72-oauth`, `php72-opcache`, `php72-pcntl`, `php72-pdo-dblib`, `php72-pdo-pgsql`, `php72-phalcon`, `php72-pspell`, `php72-rdkafka`, `php72-redis`, `php72-snmp`, `php72-ssh2`, `php72-stats`, `php72-swoole`, `php72-tidy`, `php72-timecop`, `php72-timezonedb`, `php72-uuid`, `php72-uv`, `php72-xxtea`, `php72-yac`, `php72-yaml`,

**Comment by romainneutron** [comment]
The ones that are not included are not yet compatible

File: Requirements/php-meta-requirement.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 class PhpMetaRequirement < HomebrewPhpRequirement
   if $supported_php_versions.nil?
-    $supported_php_versions = %w{php53 php54 php55 php56 php71 php70}
+    $supported_php_versions = %w{php53 php54 php55 php56 php71 php70 php72}
```
Comment (alcaeus) [inline Requirements/php-meta-requirement.rb]: Line 23 declares the last version as default if no other is installed, which shouldn't be an alpha version (which is why php71 is not in last place). The line should read `$supported_php_versions = %w{php53 php54 php55 php56 php70 php72 php71}`, making 7.1 the default if no version has been installed yet.

**Comment by javian** [comment]
This is good stuff - thanks @romainneutron. We can not however do it like this since the CI system is not designed to take this much data at the same time. I would suggest doing the core formula first with the related core modules and see if that passes. Once that has been merged we can start looking at adding the other but putting them all in one pile won't work and the best thing would be to add them separately to not overload the bottle build system. I would also urge you to include the mcrypt formula which is deprecated but now lives outside of the core code and in pecl since I'm assuming that's going to be a popular package.

**Comment by romainneutron** [comment]
Oh ok, I understand. Give me some time to automate PR splitting

**Comment by romainneutron** [comment]
@javian PR is ready with only PHP formula and core modules

**Comment by javian** [comment]
Thanks @romainneutron !

**Comment by alcaeus** [comment]
@romainneutron will you be adding the formulae mentioned above in separate PRs?

**Pull request closed without merging**